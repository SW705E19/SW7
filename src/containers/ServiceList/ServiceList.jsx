import React from 'react';
import {Grid, Card, CardContent, CardMedia, Typography, withStyles} from '@material-ui/core';

const styles = {
	root: {
		width: '100%',
	},
	card: {
		height: '20em',
		cursor:'pointer'
	},
	CardContent: {
		height: '20%',
	},
	media: {
		height: '70%',
		objectFit: 'fill'
	}
};

function ServiceList(props) {
	const classes = props.classes;
	return (
		<Grid container spacing={3} className={classes.root}>
			{	
				props.services ? 
					props.services.map( (service, i) => {
						return <Grid item xs={12} sm={4} md={3} key={i}> 
							<Card
								className={classes.card}
								onClick={() => props.onClick(service.id)}
							>
								<CardMedia 
									className={classes.media}
									component='img'
									image={'https://picsum.photos/200/300?random=' + + i}
								/>
								<CardContent
									className={classes.cardContent} 
								>
									<Typography gutterBottom variant='h5' component='h2'>
										{service.name}
									</Typography>
									<Typography
										variant="body2"
										component="p"
									>
										{service.tutorInfo.user.firstName + ' ' + service.tutorInfo.user.lastName}
									</Typography>
								</CardContent>

							</Card>
						</Grid>;
					}):
					null
			}
		</Grid>
	);
}

export default withStyles(styles)(ServiceList);
