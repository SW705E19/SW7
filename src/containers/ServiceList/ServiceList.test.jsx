import { mount, configure } from 'enzyme';
import React from 'react';
import ServiceList from './ServiceList';
import Adapter from 'enzyme-adapter-react-16';
import { Card } from '@material-ui/core';

configure({adapter: new Adapter()});

describe('<ServiceList />', () => {
	const service = {
		name: 'test',
		id: 1,
		tutorInfo: {
			user: {
				firstName: 'testtur',
				lastName: 'testtursen'
			}
		}
	};
	const services = Array(5).fill(service);
	let wrapper;
	beforeEach(() => {
		wrapper = mount(<ServiceList />);
	});

	it('ServiceList', () => {
		expect(ServiceList).toBeDefined();
	});

	it('should render correctly', () => {
		wrapper.setProps({
			servicesPerLine: 4,
			services: services
		});
		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('should render five Card components', () => {
		wrapper.setProps({
			services: services,
			servicesPerLine: 4
		});
		expect(wrapper.find(Card)).toHaveLength(5);
	});

	it('should not render any cards when undefined', () => {
		expect(wrapper.find(Card)).toHaveLength(0);
	});

	it('onChange is called when card is clicked', () => {
		const mockCallBack = jest.fn();
		wrapper.setProps({
			services: [service],
			servicesPerLine: 1,
			onClick: mockCallBack

		});
		wrapper.find(Card).prop('onClick')();
		expect(mockCallBack).toHaveBeenCalled();
	});

	it('onChange is called with serviceId as parameter', () => {
		const mockCallBack = jest.fn();
		wrapper.setProps({
			services: [service],
			servicesPerLine: 1,
			onClick: mockCallBack
		});
		const serviceId = service.id;
		wrapper.find(Card).prop('onClick')();
		expect(mockCallBack).toHaveBeenCalledWith(serviceId);
	});
});
