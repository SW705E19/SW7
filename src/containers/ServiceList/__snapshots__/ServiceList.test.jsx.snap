// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ServiceList /> should render correctly 1`] = `
"<WithStyles(ServiceList) services={{...}}>
  <ServiceList classes={{...}} services={{...}}>
    <WithStyles(ForwardRef(Grid)) container={true} spacing={3} className=\\"ServiceList-root-1\\">
      <ForwardRef(Grid) classes={{...}} container={true} spacing={3} className=\\"ServiceList-root-1\\">
        <div className=\\"MuiGrid-root ServiceList-root-1 MuiGrid-container MuiGrid-spacing-xs-3\\">
          <WithStyles(ForwardRef(Grid)) item={true} xs={12} sm={4} md={3}>
            <ForwardRef(Grid) classes={{...}} item={true} xs={12} sm={4} md={3}>
              <div className=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3\\">
                <WithStyles(ForwardRef(Card)) className=\\"ServiceList-card-2\\" onClick={[Function: onClick]}>
                  <ForwardRef(Card) classes={{...}} className=\\"ServiceList-card-2\\" onClick={[Function: onClick]}>
                    <WithStyles(ForwardRef(Paper)) className=\\"MuiCard-root ServiceList-card-2\\" elevation={1} onClick={[Function: onClick]}>
                      <ForwardRef(Paper) classes={{...}} className=\\"MuiCard-root ServiceList-card-2\\" elevation={1} onClick={[Function: onClick]}>
                        <div className=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root ServiceList-card-2 MuiPaper-rounded\\" onClick={[Function: onClick]}>
                          <WithStyles(ForwardRef(CardMedia)) className=\\"ServiceList-media-4\\" component=\\"img\\" image=\\"https://picsum.photos/200/300?random=0\\">
                            <ForwardRef(CardMedia) classes={{...}} className=\\"ServiceList-media-4\\" component=\\"img\\" image=\\"https://picsum.photos/200/300?random=0\\">
                              <img className=\\"MuiCardMedia-root ServiceList-media-4 MuiCardMedia-media MuiCardMedia-img\\" style={[undefined]} src=\\"https://picsum.photos/200/300?random=0\\" />
                            </ForwardRef(CardMedia)>
                          </WithStyles(ForwardRef(CardMedia))>
                          <WithStyles(ForwardRef(CardContent)) className={[undefined]}>
                            <ForwardRef(CardContent) classes={{...}} className={[undefined]}>
                              <div className=\\"MuiCardContent-root\\">
                                <WithStyles(ForwardRef(Typography)) gutterBottom={true} variant=\\"h5\\" component=\\"h2\\">
                                  <ForwardRef(Typography) classes={{...}} gutterBottom={true} variant=\\"h5\\" component=\\"h2\\">
                                    <h2 className=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom\\">
                                      test
                                    </h2>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                                <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" component=\\"p\\">
                                  <ForwardRef(Typography) classes={{...}} variant=\\"body2\\" component=\\"p\\">
                                    <p className=\\"MuiTypography-root MuiTypography-body2\\">
                                      testtur testtursen
                                    </p>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                              </div>
                            </ForwardRef(CardContent)>
                          </WithStyles(ForwardRef(CardContent))>
                        </div>
                      </ForwardRef(Paper)>
                    </WithStyles(ForwardRef(Paper))>
                  </ForwardRef(Card)>
                </WithStyles(ForwardRef(Card))>
              </div>
            </ForwardRef(Grid)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid)) item={true} xs={12} sm={4} md={3}>
            <ForwardRef(Grid) classes={{...}} item={true} xs={12} sm={4} md={3}>
              <div className=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3\\">
                <WithStyles(ForwardRef(Card)) className=\\"ServiceList-card-2\\" onClick={[Function: onClick]}>
                  <ForwardRef(Card) classes={{...}} className=\\"ServiceList-card-2\\" onClick={[Function: onClick]}>
                    <WithStyles(ForwardRef(Paper)) className=\\"MuiCard-root ServiceList-card-2\\" elevation={1} onClick={[Function: onClick]}>
                      <ForwardRef(Paper) classes={{...}} className=\\"MuiCard-root ServiceList-card-2\\" elevation={1} onClick={[Function: onClick]}>
                        <div className=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root ServiceList-card-2 MuiPaper-rounded\\" onClick={[Function: onClick]}>
                          <WithStyles(ForwardRef(CardMedia)) className=\\"ServiceList-media-4\\" component=\\"img\\" image=\\"https://picsum.photos/200/300?random=1\\">
                            <ForwardRef(CardMedia) classes={{...}} className=\\"ServiceList-media-4\\" component=\\"img\\" image=\\"https://picsum.photos/200/300?random=1\\">
                              <img className=\\"MuiCardMedia-root ServiceList-media-4 MuiCardMedia-media MuiCardMedia-img\\" style={[undefined]} src=\\"https://picsum.photos/200/300?random=1\\" />
                            </ForwardRef(CardMedia)>
                          </WithStyles(ForwardRef(CardMedia))>
                          <WithStyles(ForwardRef(CardContent)) className={[undefined]}>
                            <ForwardRef(CardContent) classes={{...}} className={[undefined]}>
                              <div className=\\"MuiCardContent-root\\">
                                <WithStyles(ForwardRef(Typography)) gutterBottom={true} variant=\\"h5\\" component=\\"h2\\">
                                  <ForwardRef(Typography) classes={{...}} gutterBottom={true} variant=\\"h5\\" component=\\"h2\\">
                                    <h2 className=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom\\">
                                      test
                                    </h2>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                                <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" component=\\"p\\">
                                  <ForwardRef(Typography) classes={{...}} variant=\\"body2\\" component=\\"p\\">
                                    <p className=\\"MuiTypography-root MuiTypography-body2\\">
                                      testtur testtursen
                                    </p>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                              </div>
                            </ForwardRef(CardContent)>
                          </WithStyles(ForwardRef(CardContent))>
                        </div>
                      </ForwardRef(Paper)>
                    </WithStyles(ForwardRef(Paper))>
                  </ForwardRef(Card)>
                </WithStyles(ForwardRef(Card))>
              </div>
            </ForwardRef(Grid)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid)) item={true} xs={12} sm={4} md={3}>
            <ForwardRef(Grid) classes={{...}} item={true} xs={12} sm={4} md={3}>
              <div className=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3\\">
                <WithStyles(ForwardRef(Card)) className=\\"ServiceList-card-2\\" onClick={[Function: onClick]}>
                  <ForwardRef(Card) classes={{...}} className=\\"ServiceList-card-2\\" onClick={[Function: onClick]}>
                    <WithStyles(ForwardRef(Paper)) className=\\"MuiCard-root ServiceList-card-2\\" elevation={1} onClick={[Function: onClick]}>
                      <ForwardRef(Paper) classes={{...}} className=\\"MuiCard-root ServiceList-card-2\\" elevation={1} onClick={[Function: onClick]}>
                        <div className=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root ServiceList-card-2 MuiPaper-rounded\\" onClick={[Function: onClick]}>
                          <WithStyles(ForwardRef(CardMedia)) className=\\"ServiceList-media-4\\" component=\\"img\\" image=\\"https://picsum.photos/200/300?random=2\\">
                            <ForwardRef(CardMedia) classes={{...}} className=\\"ServiceList-media-4\\" component=\\"img\\" image=\\"https://picsum.photos/200/300?random=2\\">
                              <img className=\\"MuiCardMedia-root ServiceList-media-4 MuiCardMedia-media MuiCardMedia-img\\" style={[undefined]} src=\\"https://picsum.photos/200/300?random=2\\" />
                            </ForwardRef(CardMedia)>
                          </WithStyles(ForwardRef(CardMedia))>
                          <WithStyles(ForwardRef(CardContent)) className={[undefined]}>
                            <ForwardRef(CardContent) classes={{...}} className={[undefined]}>
                              <div className=\\"MuiCardContent-root\\">
                                <WithStyles(ForwardRef(Typography)) gutterBottom={true} variant=\\"h5\\" component=\\"h2\\">
                                  <ForwardRef(Typography) classes={{...}} gutterBottom={true} variant=\\"h5\\" component=\\"h2\\">
                                    <h2 className=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom\\">
                                      test
                                    </h2>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                                <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" component=\\"p\\">
                                  <ForwardRef(Typography) classes={{...}} variant=\\"body2\\" component=\\"p\\">
                                    <p className=\\"MuiTypography-root MuiTypography-body2\\">
                                      testtur testtursen
                                    </p>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                              </div>
                            </ForwardRef(CardContent)>
                          </WithStyles(ForwardRef(CardContent))>
                        </div>
                      </ForwardRef(Paper)>
                    </WithStyles(ForwardRef(Paper))>
                  </ForwardRef(Card)>
                </WithStyles(ForwardRef(Card))>
              </div>
            </ForwardRef(Grid)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid)) item={true} xs={12} sm={4} md={3}>
            <ForwardRef(Grid) classes={{...}} item={true} xs={12} sm={4} md={3}>
              <div className=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3\\">
                <WithStyles(ForwardRef(Card)) className=\\"ServiceList-card-2\\" onClick={[Function: onClick]}>
                  <ForwardRef(Card) classes={{...}} className=\\"ServiceList-card-2\\" onClick={[Function: onClick]}>
                    <WithStyles(ForwardRef(Paper)) className=\\"MuiCard-root ServiceList-card-2\\" elevation={1} onClick={[Function: onClick]}>
                      <ForwardRef(Paper) classes={{...}} className=\\"MuiCard-root ServiceList-card-2\\" elevation={1} onClick={[Function: onClick]}>
                        <div className=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root ServiceList-card-2 MuiPaper-rounded\\" onClick={[Function: onClick]}>
                          <WithStyles(ForwardRef(CardMedia)) className=\\"ServiceList-media-4\\" component=\\"img\\" image=\\"https://picsum.photos/200/300?random=3\\">
                            <ForwardRef(CardMedia) classes={{...}} className=\\"ServiceList-media-4\\" component=\\"img\\" image=\\"https://picsum.photos/200/300?random=3\\">
                              <img className=\\"MuiCardMedia-root ServiceList-media-4 MuiCardMedia-media MuiCardMedia-img\\" style={[undefined]} src=\\"https://picsum.photos/200/300?random=3\\" />
                            </ForwardRef(CardMedia)>
                          </WithStyles(ForwardRef(CardMedia))>
                          <WithStyles(ForwardRef(CardContent)) className={[undefined]}>
                            <ForwardRef(CardContent) classes={{...}} className={[undefined]}>
                              <div className=\\"MuiCardContent-root\\">
                                <WithStyles(ForwardRef(Typography)) gutterBottom={true} variant=\\"h5\\" component=\\"h2\\">
                                  <ForwardRef(Typography) classes={{...}} gutterBottom={true} variant=\\"h5\\" component=\\"h2\\">
                                    <h2 className=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom\\">
                                      test
                                    </h2>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                                <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" component=\\"p\\">
                                  <ForwardRef(Typography) classes={{...}} variant=\\"body2\\" component=\\"p\\">
                                    <p className=\\"MuiTypography-root MuiTypography-body2\\">
                                      testtur testtursen
                                    </p>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                              </div>
                            </ForwardRef(CardContent)>
                          </WithStyles(ForwardRef(CardContent))>
                        </div>
                      </ForwardRef(Paper)>
                    </WithStyles(ForwardRef(Paper))>
                  </ForwardRef(Card)>
                </WithStyles(ForwardRef(Card))>
              </div>
            </ForwardRef(Grid)>
          </WithStyles(ForwardRef(Grid))>
          <WithStyles(ForwardRef(Grid)) item={true} xs={12} sm={4} md={3}>
            <ForwardRef(Grid) classes={{...}} item={true} xs={12} sm={4} md={3}>
              <div className=\\"MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-3\\">
                <WithStyles(ForwardRef(Card)) className=\\"ServiceList-card-2\\" onClick={[Function: onClick]}>
                  <ForwardRef(Card) classes={{...}} className=\\"ServiceList-card-2\\" onClick={[Function: onClick]}>
                    <WithStyles(ForwardRef(Paper)) className=\\"MuiCard-root ServiceList-card-2\\" elevation={1} onClick={[Function: onClick]}>
                      <ForwardRef(Paper) classes={{...}} className=\\"MuiCard-root ServiceList-card-2\\" elevation={1} onClick={[Function: onClick]}>
                        <div className=\\"MuiPaper-root MuiPaper-elevation1 MuiCard-root ServiceList-card-2 MuiPaper-rounded\\" onClick={[Function: onClick]}>
                          <WithStyles(ForwardRef(CardMedia)) className=\\"ServiceList-media-4\\" component=\\"img\\" image=\\"https://picsum.photos/200/300?random=4\\">
                            <ForwardRef(CardMedia) classes={{...}} className=\\"ServiceList-media-4\\" component=\\"img\\" image=\\"https://picsum.photos/200/300?random=4\\">
                              <img className=\\"MuiCardMedia-root ServiceList-media-4 MuiCardMedia-media MuiCardMedia-img\\" style={[undefined]} src=\\"https://picsum.photos/200/300?random=4\\" />
                            </ForwardRef(CardMedia)>
                          </WithStyles(ForwardRef(CardMedia))>
                          <WithStyles(ForwardRef(CardContent)) className={[undefined]}>
                            <ForwardRef(CardContent) classes={{...}} className={[undefined]}>
                              <div className=\\"MuiCardContent-root\\">
                                <WithStyles(ForwardRef(Typography)) gutterBottom={true} variant=\\"h5\\" component=\\"h2\\">
                                  <ForwardRef(Typography) classes={{...}} gutterBottom={true} variant=\\"h5\\" component=\\"h2\\">
                                    <h2 className=\\"MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom\\">
                                      test
                                    </h2>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                                <WithStyles(ForwardRef(Typography)) variant=\\"body2\\" component=\\"p\\">
                                  <ForwardRef(Typography) classes={{...}} variant=\\"body2\\" component=\\"p\\">
                                    <p className=\\"MuiTypography-root MuiTypography-body2\\">
                                      testtur testtursen
                                    </p>
                                  </ForwardRef(Typography)>
                                </WithStyles(ForwardRef(Typography))>
                              </div>
                            </ForwardRef(CardContent)>
                          </WithStyles(ForwardRef(CardContent))>
                        </div>
                      </ForwardRef(Paper)>
                    </WithStyles(ForwardRef(Paper))>
                  </ForwardRef(Card)>
                </WithStyles(ForwardRef(Card))>
              </div>
            </ForwardRef(Grid)>
          </WithStyles(ForwardRef(Grid))>
        </div>
      </ForwardRef(Grid)>
    </WithStyles(ForwardRef(Grid))>
  </ServiceList>
</WithStyles(ServiceList)>"
`;
